### DISCUSSION TITLE
Admin event resource type properties

### DISCUSSION THREAD

**Discussion by Wapax** [keycloak/keycloak#21644] [Q&A]
Hi everyone,

I would like to know what is the difference between `resourceTypeAsString` and `resourceType` attributes within an admin event ? It seems to have the same value...

Regards

**Comment by danielFesenmeyer** [comment]
I guess you are talking about the methods `getResourceTypeAsString() `and `getResourceType()` of class `AdminEvent`?
Those return "the same" value, as long as the resource type is a keycloak builtin resource type.
When you create your own events, for example with a resource type "dummy", `getResourceTypeAsString()` will return "dummy", but `getResourceType()` will return the Keycloak builtin enum value `ResourceType.CUSTOM`.

**Reply by Wapax** [reply]
I'm not talking especially about the methods of the `AdminEvent` interfaces but about the output I can get from the admin API.
Thank for your answer, it's more clear in my mind
